src/test/java/org/broadinstitute/hellbender/tools/walkers/annotator/ExcessHetUnitTest.java

```diff
@@ -1,253 +1,286 @@
     }
 
     @Test
     public void testAllHetsForLargeCohorts() {
-
         final int numGTs = 1000000;
 
         final List<Genotype> singletonGTs = new ArrayList<>();
         for (int i = 0; i < numGTs; i++) {
-            singletonGTs.add(makeG("ref" + i, Aref, Aref, homRefPLs));
+            singletonGTs.add(makeG("ref" + i, A_REF, A_REF, HOM_REF_PL_ARRAY));
         }
 
-        singletonGTs.add(makeG("het0", Aref, T, hetPLs));
+        singletonGTs.add(makeG("het0", A_REF, T, HET_PL_ARRAY));
 
-        final VariantContext singleton = makeVC("singleton", Arrays.asList(Aref, T), singletonGTs.toArray(new Genotype[singletonGTs.size()]));
+        final VariantContext singleton = makeVC("singleton", Arrays.asList(A_REF, T), singletonGTs.toArray(new Genotype[singletonGTs.size()]));
         final double singletonValue = ExcessHet.calculateEH(singleton, singleton.getGenotypes()).getValue();
 
         final List<Genotype> allHetGTs = new ArrayList<>();
         for (int i = 0; i < numGTs; i++) {
-            allHetGTs.add(makeG("het" + i, Aref, T, hetPLs));
+            allHetGTs.add(makeG("het" + i, A_REF, T, HET_PL_ARRAY));
         }
 
-        final VariantContext allHet = makeVC("allHet", Arrays.asList(Aref, T), allHetGTs.toArray(new Genotype[allHetGTs.size()]));
+        final VariantContext allHet = makeVC("allHet", Arrays.asList(A_REF, T), allHetGTs.toArray(new Genotype[allHetGTs.size()]));
         final double hetsValue = ExcessHet.calculateEH(allHet, allHet.getGenotypes()).getValue();
 
         Assert.assertTrue(Math.abs(singletonValue) < Math.abs(hetsValue), String.format("singleton=%f allHets=%f", singletonValue, hetsValue));
 
         //Since all hets is such an extreme case and the sample size is large here, we know that the p-value should be 0
-        Assert.assertEquals(hetsValue, ExcessHet.PHRED_SCALED_MIN_P_VALUE, DELTA_PRECISION, String.format("P-value of 0 should be phred scaled to " + ExcessHet.PHRED_SCALED_MIN_P_VALUE));
+        Assert.assertEquals(hetsValue, ExcessHet.PHRED_SCALED_MIN_P_VALUE, DELTA_PRECISION, "P-value of 0 should be phred scaled to " + ExcessHet.PHRED_SCALED_MIN_P_VALUE);
     }
 
     @DataProvider(name = "smallSets")
     public Object[][] counts() {
         return new Object[][]{
-                {1, 0, 0, 0.5},
-                {1, 1, 0, 0.5},
-                {1, 1, 1, 0.7},
-                {4, 0, 0, 0.114},
-                {2, 1, 1, 0.571},
-                {0, 2, 2, 0.957},
-                {1, 1, 40, 0.982},
-                {3, 0, 39, 0.482},
+                {1, 0, 0, 1.},
+                {1, 1, 0, 1.},
+                // The following test cases are taken from Table 1 of Wigginton et al.
+                // at https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1199378/.
+                // Results were calculated using the python implementation at
```

Comment:
Another thought: I could put a Java implementation of this slower explicit calculation here in the test code. We could check this against the test cases provided in the paper, and then use it to generate expected results for the additional test cases (rather than hardcoding the results of running the python implementation). Might save future developers a bit of time, although I'm not sure how bulletproof such a Java implementation would be...I'm inclined to trust the scipy/numpy implementations of special functions a little more.

Reply:
I don't expect to have to modify this code in the future, so don't break your back bending over backwards for future developers.  I think using a public python library in a separate file is great.

Reply:
For my part, at least, I have been thinking about parameterizing the annotation to allow the exact test to be performed when an estimate of inbreeding/outcrossing rate is available; it's mentioned near the end of Wigginton et al. as an easy modification to the standard recurrence relation. So I might revisit this code shortly.

But thanks---certainly, the less non-trivial test code I have to write for future developers, the better!